scripts/ai/AIDriveStrategyFieldWorkCourse.lua

```diff
@@ -418,7 +418,12 @@ function AIDriveStrategyFieldWorkCourse:onWaypointPassed(ix, course)
             self:raiseImplements()
             self.state = self.states.ON_CONNECTING_TRACK
         end
-            self:checkTransitionFromConnectingTrack(ix, course)
+        self:checkTransitionFromConnectingTrack(ix, course)
+        --update lastWaypoint variable to keep moving the course when passing a new waypoint.
+        --only needs to be done when in show current waypoints mode to save performance
+        if self.vehicle:getCpSettings().showCourse:getValue() == CpVehicleSettings.SHOW_COURSE_CURRENT_WPS then

```

Comment:
@tn4799 If I guess correctly you only want to update the changes on the lane change, when a turn between lanes happens. For cleaner code might be better to move this into onWaypointChange() and checjk if a turn is happing. 
@pvaiko What is your option and do you have an idea related to the changes mentioned above?

Reply:
No, I want to update the waypoints every time the current waypoint changes so that the user can see in real time what the current and next couple of steps are. It doesn't matter if the driver turns or not.

The location of this code was suggested by @pvaiko in my old pull request.

Reply:
@tn4799 It's fine. I mistakenly thought the code was in a different function then ``onWaypointPassed()``.